File: chapter1/1.3_URLify.js
// Write a method to replace all spaces in a string with '%20'. You may assume that the string
// has sufficient space at the end to hold the additional characters, and that you are given the "true"
// length of the string.

function URLify(str, len){
    let s = "";
    let endSpaces = str.length - len;
    for(let i = 0; i < str.length; i++){
        if(i < (str.length - endSpaces)){
            if(str[i] === " "){
                s += "%20";
            } else{
                s += str[i];
            }
        }
    }
    return s;
}

// Test Cases:
console.log(URLify("Mr John Smith ", 13)); // => "Mr%20Joh%20Smith"
console.log(URLify("Mr John Smith ", 14)); // => "Mr%20Joh%20Smith%20"
console.log(URLify("   hi", 7)); // => "%20%20%20hi%20%20"
console.log(URLify("   hi", 3)); // => "%20hi"
console.log(URLify("", 0)); // => ""
console.log(URLify("", 2)); // => "%20%20"
console.log(URLify("hel lo", 6)); // => "hel%20lo"

File: chapter1/1.3_URLify.py
'''

Write a method to replace all spaces in a string with '%20'. You may assume that the string
has sufficient space at the end to hold the additional characters, and that you are given the "true"
length of the string.

'''


def URLify(s):
    # Trim the given string
    st = s.strip()

    # Replace All space (unicode is \\s) to %20
    string = st.replace(' ', "%20")
    return string


print(URLify("Mr John Smith   "))  # "Mr%20Joh%20Smith"
